src/xor_name.rs

```diff
@@ -262,15 +263,17 @@ impl AsRef<XorName> for XorName {
 mod tests {
     use super::*;
     use crate::routing_table::Xorable;
+    // use bincode::{deserialize, serialize};
     use maidsafe_utilities::serialisation::{deserialise, serialise};
-    use rand;
+    use rand::Rng;
     use std::cmp::Ordering;
     use unwrap::unwrap;
 
     #[test]
-    fn serialisation_xor_name() {
-        let obj_before: XorName = rand::random();
-        let data = unwrap!(serialise(&obj_before));
+    fn serialisation_xor_name() -> Result<(), RoutingError> {
```

Comment:
Still not clear that it is a better way of doing things, so far it seem to be less useful than the original.

Reply:
Yes, it's still in the air IMO. Without backtrace it is perhaps less useful. I am still pondering this one and we should have a deeper chat. It does not belong in this PR though (I will remove it). 
It seems the more complex the test the more reason to use result based returns with accurate errors. Smaller unit test, perhaps should just panic/assert (preferred) Certainly unfinished discussion here I recon.